File: source/sat/src/main/java/ch/unisg/sat/controller/http/dto/response/ArchitectureDecisionResponseDto.java
package ch.unisg.sat.controller.http.dto.response;

import ch.unisg.sat.controller.http.dto.response.ad.AlternativeAggregateResponseDto;
import ch.unisg.sat.controller.http.dto.response.ad.AlternativeResponseDto;
import ch.unisg.sat.controller.http.dto.response.ad.IssueResponseDto;
import ch.unisg.sat.core.domain.aggregate.AlternativeAggregate;
import ch.unisg.sat.core.domain.aggregate.ArchitectureDecision;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class ArchitectureDecisionResponseDto extends ResponseDto {

    public static HashMap<String, Object> toJson(ArchitectureDecision architectureDecision) {

        HashMap<String, Object> architectureDecisionDto = new HashMap<>();

        architectureDecisionDto.put("issue", IssueResponseDto.toJson(architectureDecision.getIssue()));

        List<HashMap<String, Object>> alternativeDtoList = new ArrayList<>();

        for (AlternativeAggregate alternativeAggregate : architectureDecision.getAlternativeAggregates()) {

            alternativeDtoList.add(AlternativeResponseDto.create(alternativeAggregate.getAlternative()));
        }

        architectureDecisionDto.put("alternatives", alternativeDtoList);

        return architectureDecisionDto;
    }

    public static HashMap<String, Object> toJsonVerbose(ArchitectureDecision architectureDecision) {
        HashMap<String, Object> architectureDecisionDto = new HashMap<>();

        architectureDecisionDto.put("issue", IssueResponseDto.toJson(architectureDecision.getIssue()));
        architectureDecisionDto.put("alternativeAggregates", AlternativeAggregateResponseDto.toJsonVerbose(architectureDecision.getAlternativeAggregates()));

        return architectureDecisionDto;
    }
}

File: source/frontend/src/lib/dto/response/aggregate/ArchitectureDecisionResponseDto.ts
import {fromIssueResponseDto, IssueResponseDto} from "$lib/dto/response/entity/IssueResponseDto";
import { AlternativeAggregateResponseDto, fromAlternativeAggregatesResponseDto } from "$lib/dto/response/aggregate/AlternativeAggregateResponseDto";
import {ArchitectureDecision} from "$lib/domain/aggregate/ArchitectureDecision.svelte.js";

export class ArchitectureDecisionResponseDto {
    issue: IssueResponseDto = IssueResponseDto.create();
    alternativeAggregates : AlternativeAggregateResponseDto[] = [];
}

export function fromArchitecturalDecisionsResponseDto(architecturalDecisionsResponseDto : ArchitectureDecisionResponseDto[]) : ArchitectureDecision[] {
    return architecturalDecisionsResponseDto.map(fromArchitectureDecisionResponseDto);
}

export function fromArchitectureDecisionResponseDto(architectureDecisionResponseDto : ArchitectureDecisionResponseDto) : ArchitectureDecision {
    const architectureDecision : ArchitectureDecision = ArchitectureDecision.create();

    architectureDecision.issue = fromIssueResponseDto(architectureDecisionResponseDto.issue);
    architectureDecision.alternatives = fromAlternativeAggregatesResponseDto(architectureDecisionResponseDto.alternativeAggregates);

    return architectureDecision;

}